services/impl/src/main/java/org/exoplatform/automatic/translation/rest/AutomaticTranslationRestService.java

```diff
@@ -53,92 +58,83 @@ public AutomaticTranslationRestService(AutomaticTranslationService automaticTran
-      }
-  )
+  @ApiOperation(value = "Set Automatic translation active connector", httpMethod = "PUT", response = Response.class, produces = MediaType.APPLICATION_JSON)
+  @ApiResponses(value = { @ApiResponse(code = HTTPStatus.OK, message = "Request fulfilled"),
+      @ApiResponse(code = HTTPStatus.BAD_REQUEST, message = "Invalid Query Input"),
+      @ApiResponse(code = HTTPStatus.FORBIDDEN, message = "Not authorized to change the connector"),
+      @ApiResponse(code = HTTPStatus.INTERNAL_ERROR, message = "Internal Server Error") })
   public Response setActiveConnector(@ApiParam(value = "Connector name", required = true) @QueryParam("connector") String connector) {
     connector = (connector == null || connector.equals("null")) ? "" : connector;
-    if (automaticTranslationService.setActiveConnector(connector)) {
-      return Response.ok().build();
-    } else {
-      return Response.status(403).build();
+    try {
+      automaticTranslationService.setActiveConnector(connector);
+    } catch (RuntimeException e) {
+      LOG.error("Unable to set active connector", e);
+      return Response.status(HTTPStatus.BAD_REQUEST).entity(e.getMessage()).build();
     }
-
+    return Response.ok().build();
   }
 
   @PUT
   @Path("/setApiKey")
   @RolesAllowed("administrators")
   @Consumes(MediaType.APPLICATION_JSON)
-  @ApiOperation(
-      value = "Set Automatic translation API key",
-      httpMethod = "PUT",
-      response = Response.class,
-      produces = MediaType.APPLICATION_JSON
-  )
-  @ApiResponses(
-      value = {
-          @ApiResponse(code = 200, message = "Request fulfilled"),
-          @ApiResponse(code = 403, message = "Not authorized to change the connector")
-      }
-  )
+  @ApiOperation(value = "Set Automatic translation API key", httpMethod = "PUT", response = Response.class, produces = MediaType.APPLICATION_JSON)
+  @ApiResponses(value = { @ApiResponse(code = HTTPStatus.OK, message = "Request fulfilled"),
+      @ApiResponse(code = HTTPStatus.BAD_REQUEST, message = "Invalid Query Input"),
+      @ApiResponse(code = HTTPStatus.FORBIDDEN, message = "Not authorized to change the apiKey"),
+      @ApiResponse(code = HTTPStatus.INTERNAL_ERROR, message = "Internal Server Error") })
   public Response setApiKey(@ApiParam(value = "Connector name", required = true) @QueryParam("connector") String connector,
                             @ApiParam(value = "Api Key", required = true) @QueryParam("apikey") String apikey) {
     connector = (connector == null || connector.equals("null")) ? "" : connector;
-    if (automaticTranslationService.setApiKey(connector,apikey)) {
-      return Response.ok().build();
+    try {
+      automaticTranslationService.setApiKey(connector, apikey);
+    } catch (RuntimeException e) {
+      LOG.error("Unable to set api key connector", e);
+      return Response.status(HTTPStatus.BAD_REQUEST).entity(e.getMessage()).build();
+    }
+    return Response.ok().build();
```

Comment:
```suggestion
    return Response.noContent().build();
```
use HTTP 204 (No Content) for empty response

Reply:
done